### PR TITLE
Feature/ml 185 vectorise trivial functions

### PR DESCRIPTION
**Pull request by markos**
Vectorise trivial functions

Min: DONE: (float, int32, fp32, double, int64, fp64)
Max: DONE: (float, int32, fp32, double, int64, fp64)

Uses ParallelDispatcher where possible (when array.size() > array.data().padded_size())

### CODE REVIEW

File: libs/math/include/math/matrix_operations.hpp [hunk 1]
```diff
@@ -274,6 +262,14 @@ T Product(std::vector<T> const &obj1)
...
 meta::IfIsMathArray<ArrayType, void> Max(ArrayType const &array, typename ArrayType::Type &ret)
 {
+  // if (array.size() >= array.data().padded_size())
+  // {
+  //   ret = array.data().in_parallel().Reduce(
+  //       [](auto const &a, auto const &b) { return fetch::vectorise::Max(a, b); },
+  //       [](auto const &a) { return fetch::vectorise::Max(a); });
+  // }
+  // else
+  // {
```
Comment (vhavrylov-ncube) [inline libs/math/include/math/matrix_operations.hpp:272]: do we still need the commented code?

File: libs/math/include/math/matrix_operations.hpp [hunk 2]
```diff
@@ -394,6 +391,15 @@ T Max(std::vector<T> const &obj1)
...
 {
+  // if (array.size() >= array.data().padded_size())
+  // {
+  //   ret = array.data().in_parallel().Reduce(
+  //       [](auto const &a, auto const &b) { return fetch::vectorise::Min(a, b); },
+  //       [](auto const &a) { return fetch::vectorise::Min(a); },
+  //       fetch::math::numeric_max<typename ArrayType::Type>());
+  // }
+  // else
+  // {
```
Comment (vhavrylov-ncube) [inline libs/math/include/math/matrix_operations.hpp:402]: same here

**Review by vhavrylov-ncube — DISMISSED** [review]
remove commented code pls

File: libs/math/include/math/matrix_operations.hpp (hunk 1 above)
Comment (markos) [reply libs/math/include/math/matrix_operations.hpp:272]: the commented code is there until I get the vectorisations for other types as well, then it should work.

**Review by vhavrylov-ncube — COMMENTED** [review]
no objections on commented code

**Review by baykaner — APPROVED** [review]

**Pull request merged**